app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def picture_params
+    params.require(:picture).permit(:name, :image)
+  end
+
+  def create
+    @picture = Picture.create(picture_params)
```

Comment:
A little dangerous - what happens if something goes wrong when committing to the DB? Say you lose connection or something else. You want to use a 'guard save'

```ruby
def create
  @picture = Picture.new(picture_params)
  if @picture.save
    redirect_to '/pictures'
  else # this is what happens if it goes wrong
    flash[:notice] = @picture.errors
    redirect_to '/pictures/new'
  end
end
```